### PR TITLE
Teleport and delete buttons in objects tab.

### PR DESCRIPTION
**Pull request by VigersRay**
## About the PR

Fast travel for admins.

## Why / Balance

[QoL]

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

image

## Breaking changes

**Changelog**

I dont know how make changelog for admins.

### CODE REVIEW

**Comment by AjexRose** [comment]
make the delete button something that needs to be double or even triple clicked to function, because I know damn well im going to accidentally delete the station using this.

**Comment by VigersRay** [comment]
> please please please make the delete button something that needs to be double or even triple clicked to function, because I know damn well im going to accidentally delete the station using this.

I dont know how make this without shitcode.

**Comment by deltanedas** [comment]
better to not have an instant station deleter button than one without confirmation

**Comment by ShadowCommander** [comment]
These functions are already available in the right click menu under the admin and debug categories. Are teleport and delete needed often enough that those aren't enough?

**Comment by IProduceWidgets** [comment]
They are the reason I'm most likely to open the right click verbs. Though, I'd say I use TPHere as much as TPTo.

**Comment by VigersRay** [comment]
> These functions are already available in the right click menu under the admin and debug categories. Are teleport and delete needed often enough that those aren't enough?

Very many clicks. This may not be necessary for everyone, but in some cases it is really necessary. Well, the main argument is that there is enough space for 2 more buttons.

**Comment by VigersRay** [comment]
> These functions are already available in the right click menu under the admin and debug categories. Are teleport and delete needed often enough that those aren't enough?

Very many clicks. This may not be necessary for everyone, but in some cases it is really necessary. Well, the main argument is that there is enough space for 2 more buttons.

**Comment by VigersRay** [comment]
> They are the reason I'm most likely to open the right click verbs. Though, I'd say I use TPHere as much as TPTo.

For use tpto you ned rewrite NetId, It's tiring

**Comment by VigersRay** [comment]
> They are the reason I'm most likely to open the right click verbs. Though, I'd say I use TPHere as much as TPTo.

For use tpto you ned rewrite NetId, It's tiring

**Comment by VigersRay** [comment]
> > These functions are already available in the right click menu under the admin and debug categories. Are teleport and delete needed often enough that those aren't enough?
> 
> I could see it being useful on player tab but not really on objects.

Player tab not have space for this. Perhaps if you increase the minimum window size or remove some columns, there will be enough space.

File: Content.Client/Administration/UI/Tabs/ObjectsTab/ObjectsTab.xaml.cs [hunk 1]
```diff
@@ -27,6 +29,9 @@ public sealed partial class ObjectsTab : Control
     private readonly TimeSpan _updateFrequency = TimeSpan.FromSeconds(2);
     private TimeSpan _nextUpdate;
 
+    private Action<NetEntity>? TeleportToObjAction { get; set; }
+    private Action<NetEntity>? DeleteObjAction { get; set; }
+
```
Comment (metalgearsloth) [inline Content.Client/Administration/UI/Tabs/ObjectsTab/ObjectsTab.xaml.cs:34]: Subscribing to an object's own event makes no sense, just subscribe to it on the child.

**Pull request merged**